### PR TITLE
fix: [CDS-84710]: Allow empty value in runtime input fix for Paramete…

### PR DESCRIPTION
**Pull request by abhinavmittal-harness**
## Describe your changes

As part of this pr we are trying to make executionTarget as a Parameter Field so that they can be marked as runtime input in template and then exposed in pipeline as an run time Input.
Along with it we are deprecating onDelegate field from shell Script. We will rely on whether executionTarget is null or executionTarget.host is null.

Tested creating a template and using it in pipeline and trying combination of fixed values, runtime inputs executionTimeInputs, etc
Validated basic cases for reconciliation

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55975)

### CODE REVIEW

**Comment by abhinavmittal-harness** [comment]
trigger smartchecks

**Comment by abhinavmittal-harness** [comment]
trigger uts

**Comment by abhinavmittal-harness** [comment]
trigger commonchecks

File: 970-ng-commons/src/main/java/io/serializer/jackson/ParameterFieldDeserializer.java [hunk 1]
```diff
@@ -131,7 +131,7 @@ NGExpressionUtils.DEFAULT_INPUT_SET_EXPRESSION, extractDefaultValue(defaultValue
           : valueDeserializer.deserializeWithType(p, ctxt, valueTypeDeserializer);
       return ParameterField.createValueField(refd);
     } catch (Exception ex) {
-      if (NGExpressionUtils.NULL.equals(text)) {
+      if (NGExpressionUtils.NULL.equals(text) || NGExpressionUtils.EMPTY.equals(text)) {
```
Comment (abhinavmittal-harness) [inline 970-ng-commons/src/main/java/io/serializer/jackson/ParameterFieldDeserializer.java:134]: Add a log for an empty string and check the tpye here.
Whatever info can help identify the use

File: pipeline-service/modules/orchestration-steps/contracts/src/main/java/io/harness/steps/shellscript/ShellScriptStepInfo.java [hunk 2]
```diff
@@ -100,8 +101,9 @@ public ParameterField<List<TaskSelectorYaml>> fetchDelegateSelectors() {
   @Override
   public Map<String, ParameterField<String>> extractSecretRefs() {
     Map<String, ParameterField<String>> secretRefMap = new HashMap<>();
-    if (executionTarget != null && executionTarget.getConnectorRef() != null) {
-      secretRefMap.put("executionTarget.connectorRef", executionTarget.getConnectorRef());
+    if (ParameterField.isNotNull(executionTarget) && !"<+input>".equals(executionTarget.getExpressionValue())
+        && !ParameterField.isBlank(executionTarget.getValue().getConnectorRef())) {
```
Comment (abhinavmittal-harness) [inline pipeline-service/modules/orchestration-steps/contracts/src/main/java/io/harness/steps/shellscript/ShellScriptStepInfo.java:105]: Chekc if black could be an issue

File: pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/shellscript/ShellScriptHelperServiceImpl.java [hunk 3]
```diff
@@ -225,30 +224,39 @@ public K8sInfraDelegateConfig getK8sInfraDelegateConfig(
   @Override
   public void prepareTaskParametersForExecutionTarget(@Nonnull Ambiance ambiance,
       @Nonnull ShellScriptStepParameters shellScriptStepParameters,
-      @Nonnull ShellScriptTaskParametersNGBuilder taskParametersNGBuilder) {
-    if (!shellScriptStepParameters.onDelegate.getValue()) {
-      ExecutionTarget executionTarget = shellScriptStepParameters.getExecutionTarget();
-      validateExecutionTarget(executionTarget);
-      SSHKeySpecDTO secretSpec = (SSHKeySpecDTO) getSshKeySpec(ambiance, executionTarget);
+      @Nonnull ShellScriptTaskParametersNGBuilder taskParametersNGBuilder, boolean executeOnDelegate) {
```
Comment (abhinavmittal-harness) [inline pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/shellscript/ShellScriptHelperServiceImpl.java:227]: add a comment

Comment (abhinavmittal-harness) [reply pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/shellscript/ShellScriptHelperServiceImpl.java:227]: ParameterField.isNotNull(shellScriptStepParameters.getExecutionTarget()) is not neeeded

File: pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/shellscript/ShellScriptHelperServiceImpl.java [hunk 4]
```diff
@@ -225,30 +224,39 @@ public K8sInfraDelegateConfig getK8sInfraDelegateConfig(
...
+          .host(executionTarget.getValue().getHost().getValue());
     }
   }
 
-  private void validateExecutionTarget(ExecutionTarget executionTarget) {
-    if (executionTarget == null) {
-      throw new InvalidRequestException("Execution Target can't be empty with on delegate set to false");
+  @Override
+  public boolean toExecuteOnDelegate(@Nonnull ShellScriptStepParameters shellScriptStepParameters) {
+    if (ParameterField.isNotNull(shellScriptStepParameters.getExecutionTarget())) {
```
Comment (abhinavmittal-harness) [inline pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/shellscript/ShellScriptHelperServiceImpl.java:244]: nit: return fast

File: pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/shellscript/ShellScriptHelperServiceImpl.java [hunk 5]
```diff
@@ -310,17 +318,17 @@ public Object updateExpressions(Ambiance ambiance, Object obj) {
   }
 
   @Override
-  public String getWorkingDirectory(
-      ParameterField<String> workingDirectory, @Nonnull ScriptType scriptType, boolean onDelegate) {
-    if (workingDirectory != null && EmptyPredicate.isNotEmpty(workingDirectory.getValue())) {
-      return workingDirectory.getValue();
+  public String getWorkingDirectory(ParameterField<ExecutionTarget> executionTarget, @Nonnull ScriptType scriptType) {
+    if (ParameterField.isNotNull(executionTarget)
+        && !ParameterField.isBlank(executionTarget.getValue().getWorkingDirectory())) {
```
Comment (abhinavmittal-harness) [inline pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/shellscript/ShellScriptHelperServiceImpl.java:323]: add isNotBlack method to ParameterField

File: pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/shellscript/ShellScriptHelperServiceImpl.java [hunk 6]
```diff
@@ -310,17 +318,17 @@ public Object updateExpressions(Ambiance ambiance, Object obj) {
...
+      return executionTarget.getValue().getWorkingDirectory().getValue();
     }
     String commandPath = null;
     if (scriptType == ScriptType.BASH) {
       commandPath = "/tmp";
     } else if (scriptType == ScriptType.POWERSHELL) {
       commandPath = "%TEMP%";
-      if (onDelegate) {
+      if (ParameterField.isNull(executionTarget)) {
```
Comment (abhinavmittal-harness) [inline pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/shellscript/ShellScriptHelperServiceImpl.java:331]: add executionHost is null

File: pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/shellscript/ShellScriptHelperServiceImpl.java (hunk 5 above)
Comment (abhinavmittal-harness) [reply pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/shellscript/ShellScriptHelperServiceImpl.java:323]: Confirm gthat executonTarget is not null


Not used by command step

File: pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/shellscript/ShellScriptHelperServiceImpl.java (hunk 6 above)
Comment (abhinavmittal-harness) [reply pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/shellscript/ShellScriptHelperServiceImpl.java:331]: Refactor util method to take target and return if onDelegate is true or false and use everywhere

File: pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/shellscript/v1/ShellScriptBaseStepInfoV1.java [hunk 7]
```diff
@@ -31,7 +31,7 @@ public class ShellScriptBaseStepInfoV1 {
   ShellScriptSourceWrapperV1 source;
   ExecutionTargetV1 execution_target;
 
-  ParameterField<Boolean> on_delegate;
+  @Deprecated ParameterField<Boolean> on_delegate;
```
Comment (abhinavmittal-harness) [inline pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/shellscript/v1/ShellScriptBaseStepInfoV1.java:34]: onDelegate are being removed from now so we don't need to incorporate in yaml simplification
Let Archit know

File: 970-ng-commons/src/main/java/io/serializer/jackson/ParameterFieldDeserializer.java (hunk 1 above)
Comment (abhinavmittal-harness) [reply 970-ng-commons/src/main/java/io/serializer/jackson/ParameterFieldDeserializer.java:134]: There is not much information here that can be used to identify the use

File: pipeline-service/modules/orchestration-steps/contracts/src/main/java/io/harness/steps/shellscript/ShellScriptStepInfo.java (hunk 2 above)
Comment (abhinavmittal-harness) [reply pipeline-service/modules/orchestration-steps/contracts/src/main/java/io/harness/steps/shellscript/ShellScriptStepInfo.java:105]: Not an issue

**Comment by abhinavmittal-harness** [comment]
trigger uts

**Review by puthrayaharness — APPROVED** [review]

**Comment by brijeshdhakar** [comment]
*[`970-ng-commons/src/main/java/io/serializer/jackson/ParameterFieldDeserializer.java` line 134 at r3](https://reviewable.io/reviews/harness/harness-core/55975#-NkJPtzCAMBBzAtr3gEe:-NkJPtzCAMBBzAtr3gEf:b-fy6k0a) ([raw file](https://github.com/harness/harness-core/blob/9edfd1b9ee842bc1d78ebfe0a90105e624469690/970-ng-commons/src/main/java/io/serializer/jackson/ParameterFieldDeserializer.java#L134)):*
> ```Java
>       return ParameterField.createValueField(refd);
>     } catch (Exception ex) {
>       if (NGExpressionUtils.NULL.equals(text) || NGExpressionUtils.EMPTY.equals(text)) {
> ```

We can use EmptyPredicate.isEmpty. It checks for null and empty both.

**Review by abhinavmittal-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55975)* status: 20 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`970-ng-commons/src/main/java/io/serializer/jackson/ParameterFieldDeserializer.java` line 134 at r3](https://reviewable.io/reviews/harness/harness-core/55975#-NkJPtzCAMBBzAtr3gEe:-NkJSYX16nIIU7LDbkkz:bkrydhq) ([raw file](https://github.com/harness/harness-core/blob/9edfd1b9ee842bc1d78ebfe0a90105e624469690/970-ng-commons/src/main/java/io/serializer/jackson/ParameterFieldDeserializer.java#L134)):*
<details><summary><i>Previously, brijeshdhakar (Brijesh Dhakar) wrote…</i></summary><blockquote>

We can use EmptyPredicate.isEmpty. It checks for null and empty both.
</blockquote></details>

The check is for "NULL" string not null value

**Comment by abhinavmittal-harness** [comment]
trigger uts

**Review by brijeshdhakar — COMMENTED** [review]
> Reviewed 3 of 14 files at r1, 3 of 11 files at r2, 1 of 5 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55975)* status: 20 file(s) to review. 0 of 2 approvals obtained.

**Comment by sahilhindwani-harness** [comment]
*[`970-ng-commons/src/main/java/io/serializer/jackson/ParameterFieldDeserializer.java` line 134 at r2](https://reviewable.io/reviews/harness/harness-core/55975#gh-1406330275:-NkKWbRg9PtdZMGUMVMx:b-hqnrn7) ([raw file](https://github.com/harness/harness-core/blob/a96f6615386f249e248866be60fd97873d68ba00/970-ng-commons/src/main/java/io/serializer/jackson/ParameterFieldDeserializer.java#L134)):*
<details><summary><i>Previously, abhinavmittal-harness (Abhinav Mittal) wrote…</i></summary><blockquote>

There is not much information here that can be used to identify the use
</blockquote></details>

Why do we need this change?

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 2 of 14 files at r1, 1 of 11 files at r2, 2 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55975)* status: 20 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`970-ng-commons/src/main/java/io/harness/pms/yaml/ParameterField.java` line 171 at r4](https://reviewable.io/reviews/harness/harness-core/55975#-NkOd5w73eIbDVIfGEFw:-NkOd5w83sn5vMhUI9d2:b-2pfjt5) ([raw file](https://github.com/harness/harness-core/blob/777e04b552809f64ddf45e7e27ff3d550530d2b7/970-ng-commons/src/main/java/io/harness/pms/yaml/ParameterField.java#L171)):*
> ```Java
>   }
> 
>   public static boolean isNotBlank(ParameterField<?> actualField) {
> ```

why is this required?
___
*[`970-ng-commons/src/main/java/io/serializer/jackson/ParameterFieldDeserializer.java` line 134 at r2](https://reviewable.io/reviews/harness/harness-core/55975#gh-1406330275:-NkOdDbL-2mCnw3e6Ies:b32v57) ([raw file](https://github.com/harness/harness-core/blob/a96f6615386f249e248866be60fd97873d68ba00/970-ng-commons/src/main/java/io/serializer/jackson/ParameterFieldDeserializer.java#L134)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

Why do we need this change?
</blockquote></details>

+1
___
*[`pipeline-service/modules/orchestration/src/main/java/io/harness/engine/execution/ExecutionInputServiceHelper.java` line 71 at r2](https://reviewable.io/reviews/harness/harness-core/55975#gh-1406369129:-NkOdLAp8Zc7ZHCdinpE:briox9a) ([raw file](https://github.com/harness/harness-core/blob/a96f6615386f249e248866be60fd97873d68ba00/pipeline-service/modules/orchestration/src/main/java/io/harness/engine/execution/ExecutionInputServiceHelper.java#L71)):*
<details><summary><i>Previously, abhinavmittal-harness (Abhinav Mittal) wrote…</i></summary><blockquote>

add a log line
</blockquote></details>

@abhinavmittal-harness can we walk through the changes for this?
___
*[`pipeline-service/modules/orchestration/src/main/java/io/harness/engine/execution/ExecutionInputServiceHelper.java` line 95 at r4](https://reviewable.io/reviews/harness/harness-core/55975#-NkOdpaH1lZPceZtR6ZG:-NkOdpaH1lZPceZtR6ZH:b-r3hjn7) ([raw file](https://github.com/harness/harness-core/blob/777e04b552809f64ddf45e7e27ff3d550530d2b7/pipeline-service/modules/orchestration/src/main/java/io/harness/engine/execution/ExecutionInputServiceHelper.java#L95)):*
> ```Java
>   }
> 
>   private Map<String, Object> getExecutionInputSubMap(Map<FQN, Object> fullUserInputMap, FQN key) {
> ```

where is this method used?
___
*[`pipeline-service/modules/orchestration-steps/contracts/src/main/java/io/harness/steps/shellscript/ShellScriptBaseStepInfo.java` line 50 at r4](https://reviewable.io/reviews/harness/harness-core/55975#-NkOdvmBBXmkk6tHuYzT:-NkOdvmE6gKiuddidTkd:bsak92n) ([raw file](https://github.com/harness/harness-core/blob/777e04b552809f64ddf45e7e27ff3d550530d2b7/pipeline-service/modules/orchestration-steps/contracts/src/main/java/io/harness/steps/shellscript/ShellScriptBaseStepInfo.java#L50)):*
> ```Java
>   @NotNull ShellType shell;
>   @NotNull ShellScriptSourceWrapper source;
>   @YamlSchemaTypes({runtimeEmptyStringAllowed}) ParameterField<ExecutionTarget> executionTarget;
> ```

werent we using null instead of empty string?
___
*[`pipeline-service/modules/orchestration-steps/contracts/src/main/java/io/harness/steps/shellscript/ShellScriptStepInfo.java` line 104 at r2](https://reviewable.io/reviews/harness/harness-core/55975#gh-1406331586:-NkOeOdlB43pHtjM042A:b7wyso2) ([raw file](https://github.com/harness/harness-core/blob/a96f6615386f249e248866be60fd97873d68ba00/pipeline-service/modules/orchestration-steps/contracts/src/main/java/io/harness/steps/shellscript/ShellScriptStepInfo.java#L104)):*
<details><summary><i>Previously, abhinavmittal-harness (Abhinav Mittal) wrote…</i></summary><blockquote>

Constant for this is not there
</blockquote></details>

Use matchesInputSetPattern in NGExpressionUtils
___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/internal/ShellScriptStepFilterJsonCreatorV2.java` line 48 at r4](https://reviewable.io/reviews/harness/harness-core/55975#-NkOjXzU3L6UWKWtCmpw:-NkOjXzU3L6UWKWtCmpx:b9m03sm) ([raw file](https://github.com/harness/harness-core/blob/777e04b552809f64ddf45e7e27ff3d550530d2b7/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/internal/ShellScriptStepFilterJsonCreatorV2.java#L48)):*
> ```Java
>     //
>     //      if (ParameterField.isBlank(scriptStepInfo.getExecutionTarget().getValue().getConnectorRef())) {
>     //        throw new InvalidYamlRuntimeException(format(
> ```

why these commented code?
___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/shellscript/v1/ShellScriptBaseStepInfoV1.java` line 34 at r2](https://reviewable.io/reviews/harness/harness-core/55975#gh-1406364315:-NkOlKoV0by9iSuDyRIM:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/a96f6615386f249e248866be60fd97873d68ba00/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/shellscript/v1/ShellScriptBaseStepInfoV1.java#L34)):*
<details><summary><i>Previously, abhinavmittal-harness (Abhinav Mittal) wrote…</i></summary><blockquote>

onDelegate are being removed from now so we don't need to incorporate in yaml simplification
Let Archit know
</blockquote></details>

done

**Review by archit-harness — COMMENTED** [review]
> Reviewed 2 of 14 files at r1, 2 of 11 files at r2, 1 of 5 files at r3.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55975)* status: 16 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained.

**Review by archit-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55975)* status: 16 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/modules/orchestration-steps/contracts/src/main/java/io/harness/steps/shellscript/ShellScriptBaseStepInfo.java` line 43 at r4](https://reviewable.io/reviews/harness/harness-core/55975#-NkYpIbS3NuCyPXwPXTW:-NkYpIbS3NuCyPXwPXTX:b-azsd4y) ([raw file](https://github.com/harness/harness-core/blob/777e04b552809f64ddf45e7e27ff3d550530d2b7/pipeline-service/modules/orchestration-steps/contracts/src/main/java/io/harness/steps/shellscript/ShellScriptBaseStepInfo.java#L43)):*
> ```Java
> @AllArgsConstructor
> @TypeAlias("ShellScriptBaseStepInfo")
> public class ShellScriptBaseStepInfo {
> ```

ShellSCriptBaseStepInfo, ShellScriptStepInfo, ShellSCriptStepParams and EngineStep should be created new as inflight requests will be impacted if made change in same classes, post one release we can delete older one

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 2 of 14 files at r1, 25 of 31 files at r6, 3 of 3 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55975)* status: 34 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`pipeline-service/modules/orchestration-steps/contracts/src/main/java/io/harness/steps/shellscript/ShellScriptStepInfo.java` line 53 at r7](https://reviewable.io/reviews/harness/harness-core/55975#-NksVV1gEDLIgkF3tQ84:-NksVV1gEDLIgkF3tQ85:bdm5s2e) ([raw file](https://github.com/harness/harness-core/blob/e9caac19a5d850b3abc638987939aa5abe72685a/pipeline-service/modules/orchestration-steps/contracts/src/main/java/io/harness/steps/shellscript/ShellScriptStepInfo.java#L53)):*
> ```Java
> @OwnedBy(HarnessTeam.CDC)
> @RecasterAlias("io.harness.cdng.pipeline.stepinfo.ShellScriptStepInfo")
> public class ShellScriptStepInfo
> ```

didnt we decide to create stepInfoV0 for it?
___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/internal/ShellScriptStepFilterJsonCreatorV2.java` line 48 at r4](https://reviewable.io/reviews/harness/harness-core/55975#-NkOjXzU3L6UWKWtCmpw:-NksVt4HCump9IYu38bv:b-ctko45) ([raw file](https://github.com/harness/harness-core/blob/777e04b552809f64ddf45e7e27ff3d550530d2b7/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/internal/ShellScriptStepFilterJsonCreatorV2.java#L48)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

why these commented code?
</blockquote></details>

do we not need this code?
___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/shellscript/ShellScriptHelperServiceImpl.java` line 101 at r7](https://reviewable.io/reviews/harness/harness-core/55975#-NksW0o27x3PbQBXdOu4:-NksW0o27x3PbQBXdOu5:b-jl1jjm) ([raw file](https://github.com/harness/harness-core/blob/e9caac19a5d850b3abc638987939aa5abe72685a/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/shellscript/ShellScriptHelperServiceImpl.java#L101)):*
> ```Java
> @Slf4j
> @Deprecated
> public class ShellScriptHelperServiceImpl implements ShellScriptHelperService {
> ```

why we are deprecating complete serviceImpl, and not add in method in same class? are there too many methods?
___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/shellscript/ShellScriptStep.java` line 311 at r7](https://reviewable.io/reviews/harness/harness-core/55975#-NksWuba-W4w0yEQGCeW:-NksWuba-W4w0yEQGCeX:bctt84m) ([raw file](https://github.com/harness/harness-core/blob/e9caac19a5d850b3abc638987939aa5abe72685a/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/shellscript/ShellScriptStep.java#L311)):*
> ```Java
>   }
> 
>   private void closeLogStream(Ambiance ambiance, StepBaseParameters stepParameters) {
> ```

cant we combine closeLogStream for both old and new by passing commandUnit to it?

**Review by archit-harness — APPROVED** [review]
> Reviewed 3 of 31 files at r6.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55975)* status: 34 file(s) to review. 1 of 2 approvals obtained. 

___
*[`pipeline-service/modules/orchestration-steps/contracts/src/main/java/io/harness/steps/shellscript/ShellScriptBaseStepInfo.java` line 43 at r4](https://reviewable.io/reviews/harness/harness-core/55975#-NkYpIbS3NuCyPXwPXTW:-Nksb2FKCXXIAjApTH-W:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/777e04b552809f64ddf45e7e27ff3d550530d2b7/pipeline-service/modules/orchestration-steps/contracts/src/main/java/io/harness/steps/shellscript/ShellScriptBaseStepInfo.java#L43)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

ShellSCriptBaseStepInfo, ShellScriptStepInfo, ShellSCriptStepParams and EngineStep should be created new as inflight requests will be impacted if made change in same classes, post one release we can delete older one
</blockquote></details>

done
___
*[`pipeline-service/modules/orchestration-steps/contracts/src/main/java/io/harness/steps/shellscript/ShellScriptStepInfo.java` line 53 at r7](https://reviewable.io/reviews/harness/harness-core/55975#-NksVV1gEDLIgkF3tQ84:-Nksb38wBiufmWJiG6Mm:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/e9caac19a5d850b3abc638987939aa5abe72685a/pipeline-service/modules/orchestration-steps/contracts/src/main/java/io/harness/steps/shellscript/ShellScriptStepInfo.java#L53)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

didnt we decide to create stepInfoV0 for it?
</blockquote></details>

done
___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/internal/ShellScriptStepFilterJsonCreatorV2.java` line 48 at r4](https://reviewable.io/reviews/harness/harness-core/55975#-NkOjXzU3L6UWKWtCmpw:-NksbU429P4zLdSADyIq:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/777e04b552809f64ddf45e7e27ff3d550530d2b7/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/internal/ShellScriptStepFilterJsonCreatorV2.java#L48)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

do we not need this code?
</blockquote></details>

done
___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/shellscript/ShellScriptStep.java` line 311 at r7](https://reviewable.io/reviews/harness/harness-core/55975#-NksWuba-W4w0yEQGCeW:-Nksb9iu65ghjyX3YS_O:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/e9caac19a5d850b3abc638987939aa5abe72685a/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/shellscript/ShellScriptStep.java#L311)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

cant we combine closeLogStream for both old and new by passing commandUnit to it?
</blockquote></details>

done

**Comment by abhinavmittal-harness** [comment]
trigger uts

**Comment by abhinavmittal-harness** [comment]
trigger uts

**Review by archit-harness — DISMISSED** [review]
> Reviewed 13 of 13 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55975)* status: 35 file(s) to review. 1 of 2 approvals obtained.

**Comment by abhinavmittal-harness** [comment]
trigger uts

**Pull request merged**